xcm/pallet-xcm-benchmarks/src/generic/benchmarking.rs

```diff
@@ -306,13 +305,141 @@ benchmarks! {
 		executor.holding = holding.into();
 		let instruction = Instruction::InitiateReserveWithdraw { assets: assets_filter, reserve, xcm: Xcm(vec![]) };
 		let xcm = Xcm(vec![instruction]);
-	}:{
+	}: {
 		executor.execute(xcm)?;
 	} verify {
 		// The execute completing successfully is as good as we can check.
 		// TODO: Potentially add new trait to XcmSender to detect a queued outgoing message. #4426
 	}
 
+	burn_asset {
+		let holding = T::worst_case_holding(0);
+		let assets = holding.clone();
+
+		let mut executor = new_executor::<T>(Default::default());
+		executor.holding = holding.into();
+
+		let instruction = Instruction::BurnAsset(assets.into());
+		let xcm = Xcm(vec![instruction]);
+	}: {
+		executor.execute(xcm)?;
+	} verify {
+		assert!(executor.holding.is_empty());
+	}
+
+	expect_asset {
+		let holding = T::worst_case_holding(0);
+		let assets = holding.clone();
+
+		let mut executor = new_executor::<T>(Default::default());
+		executor.holding = holding.into();
+
+		let instruction = Instruction::ExpectAsset(assets.into());
+		let xcm = Xcm(vec![instruction]);
+	}: {
+		executor.execute(xcm)?;
+	} verify {
+		// `execute` completing successfully is as good as we can check.
+	}
+
+	expect_origin {
```

Comment:
A simple comment here that worst case scenario is when the expected origin call fails.

Reply:
I don't think it actually is -- I'm using the failure case here because it's easier to verify that it worked than the success case, since the success case looks practically identical to any other success case.

Reply:
Isn't success here means that the expected origin matches? so literally only one input could give a success output?

Reply:
Yes, but I can't do a match against the result. If we are to treat the entire XCM executor as a blackbox, then the success case is not distinguishable from the "do nothing" case, since both of them emit no observable result, and hence I've chosen the failure case.